charts/seed-bootstrap/templates/prometheus/statefulset.yaml

```diff
@@ -60,7 +60,13 @@ spec:
           successThreshold: 1
           timeoutSeconds: 3
         resources:
-{{- include "util-templates.resource-quantity" .Values.prometheus | indent 10 }}
+{{- if .Values.vpa.enabled }}
+          requests:
```

Comment:
You write that the VPA will change the requests based on observations. However, having `requests` here in the manifest will probably make Gardener overwrite the `resources` object in every seed reconciliation (so every ~30s, depending on the configured sync period).
Right? Did you see this behaviour in your tests?

Reply:
The VPA changes the `requests` of the pod spec through the mutatingwebhook. The requests of the statefulset will remain the same so gardener won't reconcile it.